### PR TITLE
feat(outputs): `app-slug` and `installation-id`

### PR DESCRIPTION
**Pull request by maboloshi**
It is convenient to use `https://api.github.com/users/$app_slug[bot]` to obtain the corresponding account ID later.
Then build `Signed-off-by: $app_slug[bot] <$id+$app_slug[bot]@users.noreply.github.com>`.

Currently, there is no Linux environment to build test snapshot files

### CODE REVIEW

**Comment by gr2m** [comment]
> Then build `Signed-off-by: $app_slug[bot] <$id+$app_slug[bot]@users.noreply.github.com>`.

that suggest you'd need the app ID as well? I'm not sure if the app ID or installation ID is needed for this to work. I know that for commits, you should use the installation ID, see this patch for example: https://github.com/octokit/core.js/commit/80c9bee698498ff3c14c95f5ac76d7f2d9d755bc.patch

`29139614` is the installation ID. [The app ID is 2740](https://api.github.com/apps/renovate)

**Comment by maboloshi** [comment]
I'm currently using the GitHub GRAPHQL api for commits, which automatically sets the author information using the account information corresponding to the token.
I currently want to build `Signed-off-by` with reference to this format.
image
Request `https://api.github.com/users/action-assistant[bot]` to get the following information. Where the `id` is the account information corresponding to the `token` produced by the github app.
```json
{
  "login": "action-assistant[bot]",
  "id": 152410706,
  "node_id": "BOT_kgDOCRWaUg",
  "avatar_url": "https://avatars.githubusercontent.com/in/659279?v=4",
  "gravatar_id": "",
  "url": "https://api.github.com/users/action-assistant%5Bbot%5D",
  "html_url": "https://github.com/apps/action-assistant",
  "followers_url": "https://api.github.com/users/action-assistant%5Bbot%5D/followers",
  "following_url": "https://api.github.com/users/action-assistant%5Bbot%5D/following{/other_user}",
  "gists_url": "https://api.github.com/users/action-assistant%5Bbot%5D/gists{/gist_id}",
  "starred_url": "https://api.github.com/users/action-assistant%5Bbot%5D/starred{/owner}{/repo}",
  "subscriptions_url": "https://api.github.com/users/action-assistant%5Bbot%5D/subscriptions",
  "organizations_url": "https://api.github.com/users/action-assistant%5Bbot%5D/orgs",
  "repos_url": "https://api.github.com/users/action-assistant%5Bbot%5D/repos",
  "events_url": "https://api.github.com/users/action-assistant%5Bbot%5D/events{/privacy}",
  "received_events_url": "https://api.github.com/users/action-assistant%5Bbot%5D/received_events",
  "type": "Bot",
  "site_admin": false,
  "name": null,
  "company": null,
  "blog": "",
  "location": null,
  "email": null,
  "hireable": null,
  "bio": null,
  "twitter_username": null,
  "public_repos": 0,
  "public_gists": 0,
  "followers": 0,
  "following": 0,
  "created_at": "2023-11-29T14:32:08Z",
  "updated_at": "2024-02-15T05:52:06Z"
}
```
`action-assistant` is the github app slug I got from the features added so far.

**Comment by maboloshi** [comment]
> > Then build `Signed-off-by: $app_slug[bot] <$id+$app_slug[bot]@users.noreply.github.com>`.
> 
> that suggest you'd need the app ID as well? I'm not sure if the app ID or installation ID is needed for this to work. I know that for commits, you should use the installation ID, see this patch for example: https://github.com/octokit/core.js/commit/80c9bee698498ff3c14c95f5ac76d7f2d9d755bc.patch
> 
> `29139614` is the installation ID. [The app ID is 2740](https://api.github.com/apps/renovate)

Similarly, the account id for this [app](https://api.github.com/apps/renovate) is `29139614`.

Request `https://api.github.com/users/renovate[bot]`
```json
{
  "login": "renovate[bot]",
  "id": 29139614,
.....
}
```
The numbers in the "from" message in this [patch](https://github.com/octokit/core.js/commit/80c9bee698498ff3c14c95f5ac76d7f2d9d755bc.patch) are consistent.
```
From 80c9bee698498ff3c14c95f5ac76d7f2d9d755bc Mon Sep 17 00:00:00 2001
From: "renovate[bot]" <29139614+renovate[bot]@users.noreply.github.com>
Date: Fri, 16 Feb 2024 16:41:56 +0000
Subject: [PATCH] chore(deps): update dependency undici to v6.6.1 [security]
....
```

**Comment by maboloshi** [comment]
The name of the github app can be changed later, but it's always possible to get the latest github app's slug using the app id and private key.
I that's one of the reasons I didn't want to hardcode `Signed-off-by` in my code.

**Comment by gr2m** [comment]
> Actually, `29139614` is not the installation ID of this [app](https://api.github.com/apps/renovate), but the account ID of the app's virtual account.

you are right, that's it, today I learned:
https://api.github.com/user/29139614

Cheers!

Yeah we can add `app-slug` as output. I still want to add `installation-id` as another output but if you don't need it we can add it later. 

Can you get the tests fixed? It might be as simple as updating the snapshots with `npx ava tests/index.js -u`

**Comment by maboloshi** [comment]
On my end, the snapshot file generated using `npx ava tests/index.js` has line breaks in the format `\r\n` instead of `\n`, and I'm guessing that after `push` the github action for `test` will still report an error.

**Review by gr2m — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -239,6 +239,14 @@ jobs:
 
 GitHub App installation access token.
 
+### `installation-id`
+
+GitHub App's installation Id.
```
Comment (parkerbxyz) [inline README.md:244]:
```suggestion
GitHub App installation ID.
```

File: README.md [hunk 2]
```diff
@@ -239,6 +239,14 @@ jobs:
...
+
+### `app-slug`
+
+GitHub App's slug.
```
Comment (parkerbxyz) [inline README.md:248]:
```suggestion
GitHub App slug.
```

File: action.yml [hunk 3]
```diff
@@ -40,6 +40,10 @@ inputs:
 outputs:
   token:
     description: "GitHub installation access token"
+  installation-id:
+    description: "GitHub App's installation Id"
```
Comment (parkerbxyz) [inline action.yml:44]:
```suggestion
    description: "GitHub App installation ID"
```

File: action.yml [hunk 4]
```diff
@@ -40,6 +40,10 @@ inputs:
...
+  app-slug:
+    description: "GitHub App's slug"
```
Comment (parkerbxyz) [inline action.yml:46]:
```suggestion
    description: "GitHub App slug"
```

File: tests/main-token-get-owner-set-to-user-fail-response.test.js [hunk 5]
```diff
@@ -5,8 +5,9 @@ await test((mockPool) => {
   process.env.INPUT_OWNER = "smockle";
   delete process.env.INPUT_REPOSITORIES;
 
-  // Mock installation id request
+  // Mock installation id and app slug request
```
Comment (parkerbxyz) [inline tests/main-token-get-owner-set-to-user-fail-response.test.js:8]:
```suggestion
  // Mock installation ID and app slug request
```

File: tests/main-token-get-owner-set-to-user-repo-unset.test.js [hunk 6]
```diff
@@ -5,8 +5,9 @@ await test((mockPool) => {
   process.env.INPUT_OWNER = "smockle";
   delete process.env.INPUT_REPOSITORIES;
 
-  // Mock installation id request
+  // Mock installation id and app slug request
```
Comment (parkerbxyz) [inline tests/main-token-get-owner-set-to-user-repo-unset.test.js:8]:
```suggestion
  // Mock installation ID and app slug request
```

File: tests/main-token-get-owner-unset-repo-unset.test.js [hunk 7]
```diff
@@ -5,8 +5,9 @@ await test((mockPool) => {
   delete process.env.INPUT_OWNER;
   delete process.env.INPUT_REPOSITORIES;
 
-  // Mock installation id request
+  // Mock installation id and app slug request
```
Comment (parkerbxyz) [inline tests/main-token-get-owner-unset-repo-unset.test.js:8]:
```suggestion
  // Mock installation ID and app slug request
```

File: tests/main.js [hunk 8]
```diff
@@ -54,8 +54,9 @@ export async function test(cb = (_mockPool) => {}, env = DEFAULT_ENV) {
 
   // Calling `auth({ type: "app" })` to obtain a JWT doesn’t make network requests, so no need to intercept.
 
-  // Mock installation id request
+  // Mock installation id and app slug request
```
Comment (parkerbxyz) [inline tests/main.js:57]:
```suggestion
  // Mock installation ID and app slug request
```

**Review by parkerbxyz — COMMENTED** [review]
Just a few small suggested changes for consistency.

**Review by parkerbxyz — APPROVED** [review]

**Comment by iamstarkov** [comment]
maybe we can put but `the account ID of the app's virtual account` in the outputs as well, as far as its already fetched?

**Comment by gr2m** [comment]
can you file a new issue? We can discuss it there. I don't see why not

**Pull request merged**